tests/unit/ml_handlers/test_spacy.py

```diff
@@ -0,0 +1,170 @@
+from unittest.mock import patch
+import pandas as pd
+
+from mindsdb_sql import parse_sql
+
+from tests.unit.executor_test_base import BaseExecutorTest
+
+
+class TestSpacy(BaseExecutorTest):
+    def run_sql(self, sql):
+        ret = self.command_executor.execute_command(parse_sql(sql, dialect="mindsdb"))
+        assert ret.error_code is None
+        if ret.data is not None:
+            columns = [col.alias if col.alias is not None else col.name for col in ret.columns]
+            return pd.DataFrame(ret.data, columns=columns)
+
+    @patch("mindsdb.integrations.handlers.postgres_handler.Handler")
+    def test_spacy_ner(self, mock_handler):
+        self.run_sql("CREATE DATABASE proj")
+
+        text = ["Apple is looking at buying U.K. startup for $1 billion"]
+        df = pd.DataFrame(text, columns=['text'])
+
+        self.set_handler(mock_handler, name="pg", tables={"df": df})
+
+        self.run_sql(
+            """
+            CREATE MODEL proj.spacy__ner__model
+            PREDICT recognition
+            USING
+              engine = 'spacy'
+              linguistic_feature = 'ner',
+              target_column = 'text';
+            """
+        )
+
+        result_df = self.run_sql(
+            """
+            SELECT recognition
+            FROM proj.spacy__ner__model
+            WHERE
+            text='Apple is looking at buying U.K. startup for $1 billion';
+            """
+        )
+
+        assert "{(28, 32, 'GPE'), (45, 55, 'MONEY'), (1, 6, 'ORG')}" in result_df["recognition"].iloc[0]
+
+    @patch("mindsdb.integrations.handlers.postgres_handler.Handler")
+    def test_spacy_lemmatization(self, mock_handler):
+        self.run_sql("CREATE DATABASE proj")
+
+        text = ["Apple is looking at buying U.K. startup for $1 billion"]
+        df = pd.DataFrame(text, columns=['text'])
+
+        self.set_handler(mock_handler, name="pg", tables={"df": df})
+
+        self.run_sql(
```

Comment:
This looks like a NER model in a lemmatization test. Is it correct?

Reply:
Correct, I will change it to lemmatization model.